flocker/testtools/test/test_base.py

```diff
@@ -0,0 +1,71 @@
+"""
+Tests for flocker base test cases.
+"""
+
+import errno
+import shutil
+
+from hypothesis import given
+from hypothesis.strategies import binary
+from testtools import TestCase
+from testtools.matchers import (
+    DirExists,
+    Not,
+    PathExists,
+)
+from testtools.testresult.doubles import Python27TestResult
+from twisted.python.filepath import FilePath
+from twisted.trial import unittest
+
+from .._base import AsyncTestCase
+
+
+class AsyncTestCaseTests(TestCase):
+
+    @given(binary())
+    def test_trial_skip_exception(self, reason):
+        """
+        If tests raise the ``SkipTest`` exported by Trial, then that's
+        recorded as a skip.
+        """
+
+        class SkippingTest(AsyncTestCase):
+            def test_skip(self):
+                raise unittest.SkipTest(reason)
+
+        test = SkippingTest('test_skip')
+        result = Python27TestResult()
```

Comment:
Right. stdlib provides poor infrastructure for testing test-related stuff. testtools is better there, but could also use some improvement there.

testtools doesn't provide a "current version of Python" test result double because the API varies hilariously by Python version, and because we try to stay out of the DWIM business by design.

I've added a comment here.